code/modules/mob/living/carbon/xenomorph/castes/spiderling/spiderling.dm

```diff
@@ -30,22 +30,49 @@
 /datum/ai_behavior/spiderling
 	target_distance = 1
 	base_action = ESCORTING_ATOM
+	var/default_escorted_atom
 
 /datum/ai_behavior/spiderling/New(loc, parent_to_assign, escorted_atom, can_heal = FALSE)
 	. = ..()
+	default_escorted_atom = escorted_atom
 	RegisterSignal(escorted_atom, COMSIG_XENOMORPH_ATTACK_LIVING, .proc/go_to_target)
 	RegisterSignal(escorted_atom, COMSIG_XENOMORPH_ATTACK_OBJ, .proc/go_to_obj_target)
 	RegisterSignal(escorted_atom, COMSIG_MOB_DEATH, .proc/spiderling_rage)
 	RegisterSignal(escorted_atom, COMSIG_LIVING_DO_RESIST, .proc/parent_resist)
 	RegisterSignal(escorted_atom, COMSIG_XENOMORPH_RESIN_JELLY_APPLIED, .proc/apply_spiderling_jelly)
 	RegisterSignal(escorted_atom, list(COMSIG_XENOMORPH_REST, COMSIG_XENOMORPH_UNREST), .proc/toggle_rest)
+	RegisterSignal(escorted_atom, COMSIG_SPIDERLING_MARK, .proc/decide_mark)
+
+/datum/ai_behavior/spiderling/proc/decide_mark(source, atom/A)
+	SIGNAL_HANDLER
+	if(!A)
+		escorted_atom = source
+		return
+	if(ishuman(A))
+		escorted_atom = null
+		INVOKE_ASYNC(src, .proc/triggered_spiderling_rage, source, A)
+		return
+	if(isobj(A))
+		var/obj/obj_target = A
+		RegisterSignal(obj_target, COMSIG_PARENT_QDELETING, .proc/only_set_escorted_atom)
+		escorted_atom = null
```

Comment:
why set escorted atom here to null isnt it about to be updated anyway

Reply:
Otherwise the spiderlings are gonna come back to widow, I need to set escorted_atom to null before sending them away